metrics/web/docs/features/navigation/diagrams/navigation_class_diagram.puml

```diff
@@ -0,0 +1,105 @@
+@startuml navigation_class_diagram
+
+class MetricsApp {}
+
+package common.presentation {
+
+    package injector.widget {
+        class InjectionContainer {}
+    }
+
+    package navigation {
+
+        package route_configration {
+            class RouteName {
+            }
+
+            class RouteConfiguration {
+                + name : RouteName
+                + path : String
+                + authorizationRequired : bool
+                + nestedRouteInformationFactory : RouteConfigurationFactory
+            }
+
+            class RouteConfigurationFactory {
+                + RouteConfiguration create()
+                - RouteConfiguration _handleNested()
+            }
+        }
+        class MetricsRouteInformationParser {
+            - RouteConfigurationFactory _routeConfigurationFactory
+            + Future<RouteConfiguration> parseRouteInformation()
+            + RouteInformation restoreRouteInformation()
+        }
+        class MetricsRouterDelegate {
+            + navigatorKey : GlobalKey<NavigatorState>
+            + currentConfiguration : RouteConfiguration
+            + navigatorObservers : List<NavigatorObservers>
+            - navigationNotifier : NavigationNotifier
+            + Future<void> setInitialRoutePath(RouteConfiguration routeConfiguration)
+            + Future<void> setNewRoutePath(RouteConfiguration routeConfiguration)
+        }
+
+        package constants {
+            class MetricsRoutes {}
+        }
+
+        package state {
+            class NavigationNotifier {
+                + pages : List<Page>
+                + currentConfiguration : RouteConfiguration
+                - _metricsPageFactory : MetricsPageFactory
+                - _routeConfigurationFactory : RouteConfigurationFactory
+                - _isUserLoggedIn : bool
+                + void handleUserStateUpdates(bool isLoggedIn)
+                + void addNewPage(Page page)
+                + void popNamed(String path)
```

Comment:
There is no `popNamed` in `Navigator` so maybe pop is enough? Also, using the `Navigator.pop` with `Navigator.onDidPop` callback set seems to be more useful as we don't have access to the current `page` from anywhere but we can pop page from anywhere.